lib/session/sessionPlugin.js

```diff
@@ -119,7 +120,7 @@ const register = function (server, options) {
                 // If we used any additional auth headers when authenticating, we need to store them in the session
                 authResponse.session.additionalAuthHeaders = null;
                 if (Object.keys(additionalAuthHeaders).length) {
-                    authResponse.session.additionalAuthHeaders = additionalAuthHeaders;
+                    authResponse.session.additionalAuthHeaders = filterAuthHeaders(additionalAuthHeaders, settings.headersToStoreInSession);
```

Comment:
Instead of filtering headers here, you can send ```settings.headersToStoreInSession``` as the ```additionalAuthHeaders``` argument

Reply:
Sujith explained that additionalAuthHeaders contains key-value pairs with already filtered headers from the request, so we need to filter it again here based on sessions headers.